dotCMS/src/main/java/com/dotcms/services/VanityUrlServices.java

```diff
@@ -195,39 +198,29 @@ public void setCachedVanityUrlList(final String siteId,
      * the commit listener related to this event is executed.
      */
     @Subscriber
+    @WrapInTransaction
     public void onCommitListener(final CommitListenerEvent commitListenerEvent) {
 
-        Contentlet contentlet = commitListenerEvent.getContentlet();
+        final Contentlet contentlet =
+                commitListenerEvent.getContentlet();
 
         try {
-            if (contentlet.isVanityUrl()) {
-
-                LocalTransaction.wrap(() -> {
-
-                    //When the contentlet finished to index we need to invalidate it on cache
-                    boolean indexed = false;
-                    try {
-                        indexed = APILocator.getContentletAPI()
-                                .isInodeIndexed(contentlet.getInode(),
-                                        contentlet.isLive(), contentlet.isWorking());
-                    } catch (DotSecurityException e) {
-                        Logger.error(this,
-                                String.format("Unable to invalidate VanityURL in cache [%s]",
-                                        contentlet.getIdentifier()), e);
-                    }
-
-                    if (indexed) {
-                        //Invalidate this VanityURL
-                        VanityUrlServices.getInstance().invalidateVanityUrl(contentlet);
-                    }
-                });
 
-            }
+            Logger.debug(this, "Invalidating the vanity: "
+                    + contentlet);
+
+            ifElse( null != contentlet && contentlet.isVanityUrl() &&
+                                    this.contentletAPI.isInodeIndexed
+                                            (contentlet.getInode(),contentlet.isLive(), contentlet.isWorking()),
+                                    () -> this.invalidateVanityUrl(contentlet),
+                                    () -> Logger.error(this,
+                                            "Unable to invalidate VanityURL in cache:" +
+                                                    contentlet) );
         } catch (Exception e) {
             Logger.error(this,
                     String.format("Unable to invalidate VanityURL in cache [%s]",
                             contentlet.getIdentifier()), e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "contentlet" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)